controllers/users.js

```diff
@@ -80,11 +80,17 @@ const getUserById = async (req, res) => {
 const getUsers = async (req, res) => {
   try {
     // getting user details by id if present.
-    const query = req.query?.query ?? "";
-    const transformedQuery = parseSearchQuery(query);
-    const qualifiers = getQualifiers(query);
-
+    const { q, dev: devParam, query } = req.query;
+    const dev = devParam === "true";
+    const queryString = (dev ? q : query) || "";
+    const transformedQuery = parseSearchQuery(queryString);
+    const qualifiers = getQualifiers(queryString);
+    // Should throw an error if the new query parameter is without feature flag
+    if (q && !dev) {
+      return res.boom.notFound("Route not found");
+    }
```

Comment:
Perhaps we could optimize the code flow by considering feature flag checks earlier in the process. Currently, we run all the functionality, including database access and user retrieval, before verifying the presence of a feature flag. This approach might have implications for performance and efficiency. It might be worth exploring ways to introduce feature flag checks at an earlier stage to improve the overall system performance and response times.

so please check this condition before.


Correct me if i written something wrong above

Reply:
The steps that are done before checking the presence of feature flag is just parsing the query string and transforming it to an object and no database access and user retrieval is done here.I think  parsing the query string and converting it to object is not going to make much difference in performance here. Please correct me if I am wrong.